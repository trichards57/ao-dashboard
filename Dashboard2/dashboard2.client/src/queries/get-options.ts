import { queryOptions } from "@tanstack/react-query";
import { notFound } from "@tanstack/react-router";

export default function getOptions<T>(uri: string, queryKey: string[]) {
  return queryOptions({
    queryKey,
    queryFn: async () => {
      const response = await fetch(uri, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      });

      if (!response.ok) {
        if (response.status === 404) {
          throw notFound();
        }
        throw new Error("Network error");
      }

      return response.json() as Promise<T>;
    },
    staleTime: 10 * 60 * 1000,
    throwOnError: true,
  });
}
