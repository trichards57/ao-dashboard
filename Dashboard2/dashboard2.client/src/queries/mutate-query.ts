import { useMutation, useQueryClient } from "@tanstack/react-query";
import { notFound } from "@tanstack/react-router";

export default function useUpdate(uri: string, queryKey: string[]) {
  const queryClient = useQueryClient();

  return function useUpdateInner<TBody>(id: string) {
    return useMutation({
      mutationFn: async (body: TBody) => {
        const response = await fetch(`${uri}/${id}`, {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify(body),
        });

        if (!response.ok) {
          if (response.status === 404) {
            // eslint-disable-next-line @typescript-eslint/no-throw-literal
            throw notFound();
          }
          throw new Error("Failed to update.");
        }
      },
      onSuccess: () => {
        queryClient.invalidateQueries({ queryKey });
      },
    });
  };
}
