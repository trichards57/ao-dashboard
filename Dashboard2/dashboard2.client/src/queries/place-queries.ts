import {
  QueryClient,
  queryOptions,
  useSuspenseQuery,
} from "@tanstack/react-query";
import { notFound } from "@tanstack/react-router";

export type Region =
  | "All"
  | "EastOfEngland"
  | "EastMidlands"
  | "London"
  | "NorthEast"
  | "NorthWest"
  | "SouthEast"
  | "SouthWest"
  | "WestMidlands";

export function districtOptions(region: Region) {
  return queryOptions({
    queryKey: ["districts", region],
    queryFn: async () => {
      if (region == "All") {
        return [];
      }

      const response = await fetch(`/api/places/${region}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      });

      if (!response.ok) {
        if (response.status === 404) {
          throw notFound();
        }
        throw new Error("Failed to fetch districts.");
      }

      return response.json() as Promise<string[]>;
    },
    staleTime: 10 * 60 * 1000,
  });
}

export function hubOptions(region: Region, district: string) {
  return queryOptions({
    queryKey: ["hubs", region, district],
    queryFn: async () => {
      if (region == "All" || district.toUpperCase() == "ALL") {
        return [];
      }

      const response = await fetch(`/api/places/${region}/${district}`, {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      });

      if (!response.ok) {
        if (response.status === 404) {
          throw notFound();
        }
        throw new Error("Failed to fetch hubs.");
      }

      return response.json() as Promise<string[]>;
    },
    staleTime: 10 * 60 * 1000,
  });
}

export function useDistricts(region: Region) {
  return useSuspenseQuery(districtOptions(region));
}

export function useHubs(region: Region, district: string) {
  return useSuspenseQuery(hubOptions(region, district));
}

export function preloadDistricts(queryClient: QueryClient, region: Region) {
  return queryClient.ensureQueryData(districtOptions(region));
}

export function preloadHubs(
  queryClient: QueryClient,
  region: Region,
  district: string,
) {
  return queryClient.ensureQueryData(hubOptions(region, district));
}
