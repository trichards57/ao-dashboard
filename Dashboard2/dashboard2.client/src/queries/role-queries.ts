import { QueryClient, useSuspenseQuery } from "@tanstack/react-query";

import getOptions from "./get-options";
import useUpdate from "./mutate-query";

export type Permission = "Deny" | "Read" | "Write";

export interface RolePermissions {
  id: string;
  name: string;
  permissions: Permission;
  vehicleConfiguration: Permission;
  vorData: Permission;
}

export interface RolePermissionsUpdate {
  permissions: Permission;
  vehicleConfiguration: Permission;
  vorData: Permission;
}

export function useUpdateRole(id: string) {
  return useUpdate("/api/roles", ["role"])<RolePermissionsUpdate>(id);
}

export function roleOptions(id: string) {
  return getOptions<RolePermissions>(`/api/roles/${id}`, ["role", id]);
}

export const allRoleOptions = getOptions<RolePermissions[]>("/api/roles", [
  "role",
]);

export function useRole(id: string) {
  return useSuspenseQuery(roleOptions(id));
}

export function useRoles() {
  return useSuspenseQuery(allRoleOptions);
}

export function preloadRole(queryClient: QueryClient, id: string) {
  return queryClient.ensureQueryData(roleOptions(id));
}

export function preloadRoles(queryClient: QueryClient) {
  return queryClient.ensureQueryData(allRoleOptions);
}
