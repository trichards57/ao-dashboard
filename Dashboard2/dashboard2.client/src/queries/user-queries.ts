import {
  useMutation,
  useQueryClient,
  UseSuspenseQueryOptions,
} from "@tanstack/react-query";
import { notFound } from "@tanstack/react-router";

export interface UserInfo {
  realName: string;
  userId: string;
  email: string;
  role: string;
  amrUsed: string;
  lastAuthenticated: string;
  otherClaims: Record<string, string>;
}

export interface UserWithRole {
  id: string;
  name: string;
  roleId: string;
  role: string;
}

export interface UserRoleUpdate {
  roleId: string;
}

export const useUpdateUser = (id: string) => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: async (user: UserRoleUpdate) => {
      const response = await fetch(`/api/users/${id}`, {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(user),
      });

      if (!response.ok) {
        if (response.status === 404) {
          throw notFound();
        }
        throw new Error("Failed to update user.");
      }
    },
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ["user"] });
    },
  });
};

export const userSettings: (
  id: string,
) => UseSuspenseQueryOptions<UserWithRole> = (id: string) => ({
  queryKey: ["user", id],
  queryFn: async () => {
    const response = await fetch(`/api/users/${id}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!response.ok) {
      if (response.status === 404) {
        throw notFound();
      }
      throw new Error("Failed to fetch user settings.");
    }

    return response.json() as Promise<UserWithRole>;
  },
  staleTime: 10 * 60 * 1000
});

export const allUserOptions: UseSuspenseQueryOptions<UserWithRole[]> = {
  queryKey: ["user"],
  queryFn: async () => {
    const response = await fetch("/api/users", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!response.ok) {
      throw new Error("Failed to fetch users.");
    }

    return response.json() as Promise<UserWithRole[]>;
  },
  staleTime: 10 * 60 * 1000
};

export const userMeOptions: UseSuspenseQueryOptions<UserInfo | null> = {
  queryKey: ["me"],
  queryFn: async () => {
    const response = await fetch("/api/users/me", {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!response.ok) {
      return Promise.resolve(null);
    }

    return response.json() as Promise<UserInfo>;
  },
  staleTime: 1000 * 60 * 5,
};
