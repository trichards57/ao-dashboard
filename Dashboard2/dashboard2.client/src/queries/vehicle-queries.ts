import { useMutation, useQueryClient } from "@tanstack/react-query";

export interface VehicleSettings {
  id: string;
  registration: string;
  callSign: string;
  hub: string;
  district: string;
  region: string;
  vehicleType: string;
  forDisposal: boolean;
}

export interface UpdateVehicleSettings {
  registration: string;
  hub: string;
  callSign: string;
  district: string;
  region: string;
  vehicleType: string;
  forDisposal: boolean;
}

export const useUpdateVehicle = () => {
  const queryClient = useQueryClient();

  return useMutation({
    mutationFn: async (vehicle: UpdateVehicleSettings) => {
      const response = await fetch("/api/vehicles", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(vehicle),
      });

      if (!response.ok) {
        throw new Error("Failed to update vehicle.");
      }
    },
    onSuccess: () => {
      queryClient.invalidateQueries({ queryKey: ["vehicle"] });
    },
  });
};

export const vehicleSettings = (id: string) => ({
  queryKey: ["vehicle", id],
  queryFn: async () => {
    const response = await fetch(`/api/vehicles/${id}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!response.ok) {
      throw new Error("Failed to fetch vehicle settings.");
    }

    return response.json() as Promise<VehicleSettings>;
  },
});

export const settingsOptions = (
  region: string | undefined,
  district: string | undefined,
  hub: string | undefined,
) => ({
  queryKey: ["settings", region ?? "All", district ?? "All", hub ?? "All"],
  queryFn: async () => {
    const response = await fetch(
      `/api/vehicles?region=${region ?? "All"}&district=${district ?? "All"}&hub=${hub ?? "All"}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      },
    );

    if (!response.ok) {
      throw new Error("Failed to fetch vehicle settings.");
    }

    return response.json() as Promise<VehicleSettings[]>;
  },
});
