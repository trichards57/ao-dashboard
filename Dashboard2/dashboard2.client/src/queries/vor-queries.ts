export interface VorStatistics {
  totalVehicles: number;
  availableVehicles: number;
  vorVehicles: number;
  pastAvailability: Record<string, number>;
}

export interface VorStatus {
  id: string;
  region: string;
  hub: string;
  district: string;
  registration: string;
  callSign: string;
  summary: string;
  isVor: boolean;
  dueBack: string;
}

export const statisticsOptions = (
  region: string,
  district: string,
  hub: string,
) => ({
  queryKey: ["statistics", region ?? "All", district ?? "All", hub ?? "All"],
  queryFn: async () => {
    const response = await fetch(
      `/api/vor/statistics?region=${region ?? "All"}&district=${district ?? "All"}&hub=${hub ?? "All"}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      },
    );

    if (!response.ok) {
      throw new Error("Failed to fetch hubs.");
    }

    return response.json() as Promise<VorStatistics>;
  },
});

export const statusOptions = (
  region: string,
  district: string,
  hub: string,
) => ({
  queryKey: ["status", region ?? "All", district ?? "All", hub ?? "All"],
  queryFn: async () => {
    const response = await fetch(
      `/api/vor?region=${region ?? "All"}&district=${district ?? "All"}&hub=${hub ?? "All"}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json",
        },
      },
    );

    if (!response.ok) {
      throw new Error("Failed to fetch hubs.");
    }

    return response.json() as Promise<VorStatus[]>;
  },
});
