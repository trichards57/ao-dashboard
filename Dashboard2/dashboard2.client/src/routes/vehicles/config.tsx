import { createFileRoute, Link } from "@tanstack/react-router";
import validatePlace, { getPlaceDeps } from "../../support/validate-place";
import PlacePicker from "../../components/place-picker";
import {
  preloadAllVehicleSettings,
  useAllVehicleSettings,
} from "../../queries/vehicle-queries";
import { useState } from "react";
import PagePicker from "../../components/page-picker";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPen } from "@fortawesome/free-solid-svg-icons";
import { regionToString } from "../../components/region-converter";
import { useTitle } from "../../components/useTitle";
import {
  preloadDistricts,
  preloadHubs,
  Region,
} from "../../queries/place-queries";
import {
  redirectIfLoggedOut,
  redirectIfNoPermission,
} from "../../support/check-logged-in";

const PageSize = 10;

export function VehicleConfig({
  region,
  district,
  hub,
}: {
  region: Region;
  district: string;
  hub: string;
}) {
  const { data } = useAllVehicleSettings(region, district, hub);
  const [page, setPage] = useState(0);

  useTitle("Vehicle Setup");

  const itemsToDisplay = data
    .toSorted((a, b) => a.callSign.localeCompare(b.callSign))
    .slice(page * PageSize, (page + 1) * PageSize);
  const showPagination = data.length > PageSize;

  return (
    <>
      <h1 className="title">Vehicle Setup</h1>
      <PlacePicker region={region} district={district} hub={hub} />
      <div className="table-container">
        <table className="table is-striped is-fullwidth">
          <thead>
            <tr>
              <th className="edit-col"></th>
              <th className="call-sign-col">Call Sign</th>
              <th className="reg-col">Registration</th>
              <th>Region</th>
              <th>District</th>
              <th>Hub</th>
            </tr>
          </thead>
          <tbody>
            {itemsToDisplay.length > 0 &&
              itemsToDisplay.map((i) => (
                <tr key={i.id}>
                  <td className="edit-col">
                    <Link to="/vehicles/edit/$id" params={{ id: i.id }}>
                      <FontAwesomeIcon icon={faPen} />
                    </Link>
                  </td>
                  <td className="call-sign-col">{i.callSign}</td>
                  <td className="reg-col">{i.registration}</td>
                  <td className="place-col">{regionToString(i.region)}</td>
                  <td className="place-col">{i.district}</td>
                  <td className="place-col">{i.hub}</td>
                </tr>
              ))}

            {itemsToDisplay.length === 0 && (
              <tr>
                <td colSpan={5} className="no-vehicles subtitle is-6">
                  No Vehicles
                </td>
              </tr>
            )}
          </tbody>
        </table>
      </div>
      {showPagination && (
        <PagePicker
          page={page}
          pages={Math.ceil(data.length / PageSize)}
          setPage={setPage}
        />
      )}
    </>
  );
}

export const Route = createFileRoute("/vehicles/config")({
  validateSearch: validatePlace,
  loaderDeps: getPlaceDeps,
  loader: ({ deps, context }) =>
    Promise.all([
      preloadAllVehicleSettings(
        context.queryClient,
        deps.region,
        deps.district,
        deps.hub,
      ),
      preloadDistricts(context.queryClient, deps.region),
      preloadHubs(context.queryClient, deps.region, deps.district),
    ]),
  component: function Component() {
    return (
      <VehicleConfig
        region={Route.useSearch().region}
        district={Route.useSearch().district}
        hub={Route.useSearch().hub}
      />
    );
  },
  beforeLoad: ({ context, search }) => {
    redirectIfLoggedOut(context);
    redirectIfNoPermission(context.canEditVehicles, search);
  },
});
