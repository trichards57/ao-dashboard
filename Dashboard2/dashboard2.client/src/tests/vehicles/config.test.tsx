import { describe, expect, it, Mock, vi, beforeEach } from "vitest";
import ReactDOM from "react-dom/client";
import { act } from "react";
import { render } from "@testing-library/react";
import { VehicleConfig } from "../../routes/vehicles/config";
import {
  useAllVehicleSettings,
  VehicleSettings,
} from "../../queries/vehicle-queries";
import { useDistricts, useHubs } from "../../queries/place-queries";

vi.mock("../../queries/vehicle-queries");
vi.mock("../../queries/place-queries");

const testItem: VehicleSettings[] = [
  {
    callSign: "WR122",
    district: "Avon",
    hub: "Bristol",
    id: "1",
    registration: "AB12 CDE",
    region: "SouthWest",
    vehicleType: "OffRoadAmbulance",
    forDisposal: false,
  },
  {
    callSign: "WR123",
    district: "Avon",
    hub: "Bristol",
    id: "2",
    registration: "AB12 HTE",
    region: "SouthWest",
    vehicleType: "FrontLineAmbulance",
    forDisposal: false,
  },
  {
    callSign: "WR124",
    district: "Avon",
    hub: "Bristol",
    id: "3",
    registration: "AB12 CFE",
    region: "SouthWest",
    vehicleType: "AllWheelDrive",
    forDisposal: false,
  },
];

const testDistricts = ["Avon", "Bedfordshire", "Berkshire", "Buckinghamshire"];
const testHubs = ["Bristol", "Bath", "Weston-super-Mare", "Yeovil"];

describe("Vehicle Config Component", () => {
  beforeEach(() => {
    (useAllVehicleSettings as Mock).mockReturnValue({
      data: testItem,
    });
    (useDistricts as Mock).mockReturnValue({
      data: testDistricts,
    });
    (useHubs as Mock).mockReturnValue({
      data: testHubs,
    });
  });

  it("renders without error", () => {
    const root = ReactDOM.createRoot(document.createElement("div"));
    expect(() => {
      act(() => {
        root.render(<VehicleConfig region="All" district="All" hub="All" />);
        root.unmount();
      });
    }).not.toThrow();
  });

  it("renders correctly", () => {
    const { asFragment } = render(
      <VehicleConfig region="SouthWest" district="Avon" hub="Bristol" />,
    );
    expect(asFragment()).toMatchSnapshot();
    expect(useAllVehicleSettings).toHaveBeenCalledWith(
      "SouthWest",
      "Avon",
      "Bristol",
    );
    expect(useDistricts).toHaveBeenCalledWith("SouthWest");
    expect(useHubs).toHaveBeenCalledWith("SouthWest", "Avon");
  });

  it("renders nothing correctly", () => {
    (useAllVehicleSettings as Mock).mockReturnValue({
      data: [],
    });

    const { asFragment } = render(
      <VehicleConfig region="SouthWest" district="Avon" hub="Bristol" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
