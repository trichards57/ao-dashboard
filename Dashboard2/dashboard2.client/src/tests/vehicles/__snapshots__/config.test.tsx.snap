// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Vehicle Config Component > renders correctly 1`] = `
<DocumentFragment>
  <h1
    class="title"
  >
    Vehicle Setup
  </h1>
  <div
    class="columns is-desktop"
  >
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="region"
        >
          Region
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="region"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="EastOfEngland"
            >
              East of England
            </option>
            <option
              value="EastMidlands"
            >
              East Midlands
            </option>
            <option
              value="London"
            >
              London
            </option>
            <option
              value="NorthEast"
            >
              North East
            </option>
            <option
              value="NorthWest"
            >
              North West
            </option>
            <option
              value="SouthEast"
            >
              South East
            </option>
            <option
              value="SouthWest"
            >
              South West
            </option>
            <option
              value="WestMidlands"
            >
              West Midlands
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="district"
        >
          District
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="region"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="Avon"
            >
              Avon
            </option>
            <option
              value="Bedfordshire"
            >
              Bedfordshire
            </option>
            <option
              value="Berkshire"
            >
              Berkshire
            </option>
            <option
              value="Buckinghamshire"
            >
              Buckinghamshire
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="hub"
        >
          Hub
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="hub"
          >
            <option
              value="all"
            >
              All
            </option>
            <option
              value="Bristol"
            >
              Bristol
            </option>
            <option
              value="Bath"
            >
              Bath
            </option>
            <option
              value="Weston-super-Mare"
            >
              Weston-super-Mare
            </option>
            <option
              value="Yeovil"
            >
              Yeovil
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    class="table-container"
  >
    <table
      class="table is-striped is-fullwidth"
    >
      <thead>
        <tr>
          <th
            class="edit-col"
          />
          <th
            class="call-sign-col"
          >
            Call Sign
          </th>
          <th
            class="reg-col"
          >
            Registration
          </th>
          <th>
            Region
          </th>
          <th>
            District
          </th>
          <th>
            Hub
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="edit-col"
          >
            <a
              params="[object Object]"
              to="/vehicles/edit/$id"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pen "
                data-icon="pen"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M362.7 19.3L314.3 67.7 444.3 197.7l48.4-48.4c25-25 25-65.5 0-90.5L453.3 19.3c-25-25-65.5-25-90.5 0zm-71 71L58.6 323.5c-10.4 10.4-18 23.3-22.2 37.4L1 481.2C-1.5 489.7 .8 498.8 7 505s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L421.7 220.3 291.7 90.3z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </td>
          <td
            class="call-sign-col"
          >
            WR122
          </td>
          <td
            class="reg-col"
          >
            AB12 CDE
          </td>
          <td
            class="place-col"
          >
            South West
          </td>
          <td
            class="place-col"
          >
            Avon
          </td>
          <td
            class="place-col"
          >
            Bristol
          </td>
        </tr>
        <tr>
          <td
            class="edit-col"
          >
            <a
              params="[object Object]"
              to="/vehicles/edit/$id"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pen "
                data-icon="pen"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M362.7 19.3L314.3 67.7 444.3 197.7l48.4-48.4c25-25 25-65.5 0-90.5L453.3 19.3c-25-25-65.5-25-90.5 0zm-71 71L58.6 323.5c-10.4 10.4-18 23.3-22.2 37.4L1 481.2C-1.5 489.7 .8 498.8 7 505s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L421.7 220.3 291.7 90.3z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </td>
          <td
            class="call-sign-col"
          >
            WR123
          </td>
          <td
            class="reg-col"
          >
            AB12 HTE
          </td>
          <td
            class="place-col"
          >
            South West
          </td>
          <td
            class="place-col"
          >
            Avon
          </td>
          <td
            class="place-col"
          >
            Bristol
          </td>
        </tr>
        <tr>
          <td
            class="edit-col"
          >
            <a
              params="[object Object]"
              to="/vehicles/edit/$id"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-pen "
                data-icon="pen"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M362.7 19.3L314.3 67.7 444.3 197.7l48.4-48.4c25-25 25-65.5 0-90.5L453.3 19.3c-25-25-65.5-25-90.5 0zm-71 71L58.6 323.5c-10.4 10.4-18 23.3-22.2 37.4L1 481.2C-1.5 489.7 .8 498.8 7 505s15.3 8.5 23.7 6.1l120.3-35.4c14.1-4.2 27-11.8 37.4-22.2L421.7 220.3 291.7 90.3z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </td>
          <td
            class="call-sign-col"
          >
            WR124
          </td>
          <td
            class="reg-col"
          >
            AB12 CFE
          </td>
          <td
            class="place-col"
          >
            South West
          </td>
          <td
            class="place-col"
          >
            Avon
          </td>
          <td
            class="place-col"
          >
            Bristol
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;

exports[`Vehicle Config Component > renders nothing correctly 1`] = `
<DocumentFragment>
  <h1
    class="title"
  >
    Vehicle Setup
  </h1>
  <div
    class="columns is-desktop"
  >
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="region"
        >
          Region
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="region"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="EastOfEngland"
            >
              East of England
            </option>
            <option
              value="EastMidlands"
            >
              East Midlands
            </option>
            <option
              value="London"
            >
              London
            </option>
            <option
              value="NorthEast"
            >
              North East
            </option>
            <option
              value="NorthWest"
            >
              North West
            </option>
            <option
              value="SouthEast"
            >
              South East
            </option>
            <option
              value="SouthWest"
            >
              South West
            </option>
            <option
              value="WestMidlands"
            >
              West Midlands
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="district"
        >
          District
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="region"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="Avon"
            >
              Avon
            </option>
            <option
              value="Bedfordshire"
            >
              Bedfordshire
            </option>
            <option
              value="Berkshire"
            >
              Berkshire
            </option>
            <option
              value="Buckinghamshire"
            >
              Buckinghamshire
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="column"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="hub"
        >
          Hub
        </label>
        <div
          class="select is-fullwidth"
        >
          <select
            id="hub"
          >
            <option
              value="all"
            >
              All
            </option>
            <option
              value="Bristol"
            >
              Bristol
            </option>
            <option
              value="Bath"
            >
              Bath
            </option>
            <option
              value="Weston-super-Mare"
            >
              Weston-super-Mare
            </option>
            <option
              value="Yeovil"
            >
              Yeovil
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    class="table-container"
  >
    <table
      class="table is-striped is-fullwidth"
    >
      <thead>
        <tr>
          <th
            class="edit-col"
          />
          <th
            class="call-sign-col"
          >
            Call Sign
          </th>
          <th
            class="reg-col"
          >
            Registration
          </th>
          <th>
            Region
          </th>
          <th>
            District
          </th>
          <th>
            Hub
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="no-vehicles subtitle is-6"
            colspan="5"
          >
            No Vehicles
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
